qiita_db/artifact.py

```diff
@@ -512,21 +512,30 @@ def delete(cls, artifact_id):
                 raise qdb.exceptions.QiitaDBArtifactDeletionError(
                     artifact_id, "it is public")
 
-            # Check if this artifact has any children
-            if instance.children:
-                raise qdb.exceptions.QiitaDBArtifactDeletionError(
-                    artifact_id,
-                    "it has children: %s"
-                    % ', '.join([str(c.id) for c in instance.children]))
+            # delete children if children exists
+            for c in sorted(instance.children):
```

Comment:
I wonder, if deleting individual children in sequence will take time, if we should have a version of delete that can accept a list of children and process the deletes on the back-end 'in bulk'. What do you think?

Reply:
IMHO, this could be made an issue for later...